magento/pwa-studio
# Refactors VeniaAdapter to improve initial page load

## Description

This PR refactors `VeniaAdapter` to improve initial page load. It:

  * Updates `VeniaAdapter` to a functional component
  * Pre-instantiates an Apollo cache object (`InMemoryCache`) at `require` time so that it doesn't have to when the component is instantiated

## Related Issue





Closes #1741 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Ensure the Venia home page loads correctly
1. Issue some GraphQL requests (navigate around the app) and ensure everything still works
1. Open Chrome Dev Tools > Application > Local Storage and clear `apollo-cache-persist` if it exists. Reload the page and ensure that `apollo-cache-persist` is recreated. This tests that we didn't break cache persistence.

## Screenshots / Screen Captures (if appropriate)

**Before this PR**

![](https://user-images.githubusercontent.com/13182778/65333560-59c9aa80-db86-11e9-9221-d98dbcc833f0.png)

**After this PR**

![Screen Shot 2019-10-01 at 10 00 50 AM](https://user-images.githubusercontent.com/13182778/65977254-367ce600-e437-11e9-9d92-df49aff52448.png)

See that the "Self Time" of `VeniaAdapter` has decreased.
This measures how long a component takes to render itself (not including its child components).

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
